Assets/_Game/Scripts/Player/PlayerMove.cs

```diff
@@ -21,7 +27,13 @@ public class PlayerMove : MonoBehaviour {
 
   private void Move() {
     Vector3 movement = new Vector3(joystick.Horizontal, joystick.Vertical, 0) * Time.deltaTime * playerLoader.speed;
-    transform.position += movement;
+    Vector3 newPosition = transform.position + movement;
+
+    newPosition.x = Mathf.Clamp(newPosition.x, pointLeft.position.x, pointRight.position.x);
+    newPosition.y = Mathf.Clamp(newPosition.y, pointDown.position.y, pointTop.position.y);
```

Comment:
we need to use Rigidbody2D position instead of transform.position and dont need to use pointLeft, pointRight, pointTop, pointDown)
https://docs.unity3d.com/ScriptReference/Rigidbody2D-position.html